### DISCUSSION TITLE
Welcome to gltf-test Discussions!

### DISCUSSION THREAD

**Discussion by cx20** [cx20/gltf-test#560] [Announcements]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by elalish** [comment]
I'm the maintainer of `<model-viewer>` and you asked in an earlier thread about our fidelity test suite: https://github.com/cx20/gltf-test/issues/554#issuecomment-1093450186

You can see how the gltf sample viewer was added to our test suite here: https://github.com/google/model-viewer/pull/1962/files#diff-6856cd1835a7703476eaf2fd91803ba39777e755d78d93c5ee484b00892479f0; really the only other addition is here: https://github.com/google/model-viewer/pull/1962/files#diff-298d04f97f7d31ba199f0c8e5fbe54c33f75c1185b01c76c94c5e366d6b13d6d

There are similar PRs adding other renderers like babylon.js and dspbr-pt. We've already set each of these to have identical IBL, tone mapping, exposure, etc. It's not generally very hard to make a new renderer match once they're brought into our framework. I'm happy to help.

**Reply by cx20** [reply]
Thank you for giving us a pointer on how to start working on this.
I would like to learn a lot about modern web development because I don't know enough about it.
As a starting point, I would like to deepen my understanding of TypeScript and Webpack.
(Maybe even those technologies are legacy, but they seem modern to me.)

**Reply by emadurandal** [reply]
@elalish Hi, I'm the author of Rhodonite library, which is one of the libraries in `gltf-test` repository, and a friend of cx20 and just recently consulted with him about improving this meaningful project.

@cx20 Since the man-hours are limited, it would be reasonable to let us use this <model-viewer> as a component of gltf-test site rather than creating a unified environment ourselves. I will look into the `model-viewer` on my end.

@elalish Also, on a different note, please let me send a pull request to `model-viewer` for my Rhodonite support.

**Reply by emadurandal** [reply]
@cx20 I think it would be mutually beneficial for both projects to let cx20's gltf-test use model-viewer and also send PRs for other renderers to the model-viewer/fidelity-test-suite.
@elalish Feel free to point out if my understanding is incorrect.

**Reply by cx20** [reply]
I just discussed the following with @emadurandal.

- @emadurandal will send a PR to the fidelity test suite for a sample `Rhodonite` `model-viewer`.
- Each gltf-test library sample code is not going to be model viewer code. Each sample code is positioned as a minimum sample that explains the library.
- I plan to add one sample code that uses model-viewer separately from the existing library.

New (modern) efforts will be started with new repositories to avoid repository bloat.
The outline of the new repository is as follows.

- Import glTF-Sample-Models as a submodule
- I think glTF-Sample-Models will refer to the forked version. New models such as unapproved gltf extensions will be added there.
- Install the latest version of the library from npm using package.json
- Sample code is based on TypeScript
- Upload the built file to GitHub Pages

**Reply by elalish** [reply]
@emadurandal Looking forward to a PR! Please let me know if you have any trouble or need any pointers. Opening a draft PR is often a good way to get help in our repo. Thanks @cx20 for your work in driving glTF rendering convergence. If there's anything we can do to collaborate and help reduce duplicated effort, please let us know.

**Reply by emadurandal** [reply]
@elalish Thank you, I will.

**Reply by emadurandal** [reply]
@elalish I’m still working on it. It is taking a bit of time due to camera alignment issues, but I will be able to send you a PR in a little while.

**Reply by emadurandal** [reply]
@elalish I sent the PR. Please review it: https://github.com/google/model-viewer/pull/3513
